### ISSUE TITLE
Vagrant cannot mount /vagrant using Vagrant v1.7.4 and Virtualbox v4.3.30

### ISSUE THREAD

**Issue by predatorian3** [mitchellh/vagrant#6175]
I'm using a custom build CentOS 6.7 64bit box that I created. I was able to install Virtualbox Guest Additions, and package the box. However, now, when I issue `vagrant up` I cannot mount the /vagrant folder.

```
C:\Users\user01\Desktop\Vagrant_Files\razor_server
λ vagrant plugin install vagrant-vbguest
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
Installed the plugin 'vagrant-vbguest (0.10.0)'!

C:\Users\user01\Desktop\Vagrant_Files\razor_server
λ vagrant halt
==> default: Attempting graceful shutdown of VM...

C:\Users\user01\Desktop\Vagrant_Files\razor_server
λ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 8080 => 8080 (adapter 1)
    default: 5432 => 5432 (adapter 1)
    default: 8150 => 8150 (adapter 1)
    default: 8151 => 8151 (adapter 1)
    default: 22 => 2222 (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
    default: Warning: Connection timeout. Retrying...
==> default: Machine booted and ready!
GuestAdditions 4.3.30 running --- OK.
==> default: Checking for guest additions in VM...
==> default: Setting hostname...
==> default: Mounting shared folders...
    default: /vagrant => C:/Users/user01/Desktop/Vagrant_Files/razor_server
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` vagrant /vagrant
mount -t vboxsf -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant

The error output from the last command was:

mount: Protocol error


C:\Users\user01\Desktop\Vagrant_Files\razor_server
λ vagrant ssh
Last login: Fri Aug 21 19:56:39 2015 from 10.0.2.2
[vagrant@razor-server ~]$ sudo /etc/init.d/vboxadd setup
Removing existing VirtualBox non-DKMS kernel modules       [  OK  ]
Building the VirtualBox Guest Additions kernel modules
Building the main Guest Additions module                   [  OK  ]
Building the shared folder support module                  [  OK  ]
Building the OpenGL support module                         [FAILED]
(Look at /var/log/vboxadd-install.log to find out what went wrong)
Doing non-kernel setup of the Guest Additions              [  OK  ]
[vagrant@razor-server ~]$ exit
logout
Connection to 127.0.0.1 closed.

C:\Users\user01\Desktop\Vagrant_Files\razor_server
λ vagrant reload
==> default: Attempting graceful shutdown of VM...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 8080 => 8080 (adapter 1)
    default: 5432 => 5432 (adapter 1)
    default: 8150 => 8150 (adapter 1)
    default: 8151 => 8151 (adapter 1)
    default: 22 => 2222 (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
==> default: Machine booted and ready!
GuestAdditions 4.3.30 running --- OK.
==> default: Checking for guest additions in VM...
==> default: Setting hostname...
==> default: Mounting shared folders...
    default: /vagrant => C:/Users/user01/Desktop/Vagrant_Files/razor_server
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` vagrant /vagrant
mount -t vboxsf -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant

The error output from the last command was:

mount: Protocol error


C:\Users\user01\Desktop\Vagrant_Files\razor_server
λ vagrant version
Installed Version: 1.7.4
Latest Version: 1.7.4

You're running an up-to-date version of Vagrant!

C:\Users\user01\Desktop\Vagrant_Files\razor_server
λ vagrant ssh
Last login: Fri Aug 21 20:17:41 2015 from 10.0.2.2
[vagrant@razor-server ~]$ sudo ln -s /opt/VBoxGuestAdditions-4.3.10/lib/VBoxGuestAdditions /usr/lib/VBoxGuestA dditions
[vagrant@razor-server ~]$ exit
logout
Connection to 127.0.0.1 closed.

C:\Users\user01\Desktop\Vagrant_Files\razor_server
λ vagrant reload
==> default: Attempting graceful shutdown of VM...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 8080 => 8080 (adapter 1)
    default: 5432 => 5432 (adapter 1)
    default: 8150 => 8150 (adapter 1)
    default: 8151 => 8151 (adapter 1)
    default: 22 => 2222 (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
    default: Warning: Connection timeout. Retrying...
==> default: Machine booted and ready!
GuestAdditions 4.3.30 running --- OK.
==> default: Checking for guest additions in VM...
==> default: Setting hostname...
==> default: Mounting shared folders...
    default: /vagrant => C:/Users/user01/Desktop/Vagrant_Files/razor_server
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` vagrant /vagrant
mount -t vboxsf -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant

The error output from the last command was:

mount: Protocol error
```

Should I open this on Virtualbox as well?

**Comment by sethvargo** [comment]
Hi @predatorian3 

Have you installed the Vbox guest additions at the same version?

**Comment by predatorian3** [comment]
I'll have to try this when I am home.

**Comment by predatorian3** [comment]
Yes, I updated Vagrant, and VirtualBox, installed the Guest Additions 4.3.30 and updated the extensions pack. Still got this.

```
C:\Users\user01\Desktop\Vagrant_Files\razor_server
λ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'CentOS_6_Minimal'...
==> default: Matching MAC address for NAT networking...
==> default: Setting the name of the VM: razor_server_default_1440550918807_99200
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 8080 => 8080 (adapter 1)
    default: 5432 => 5432 (adapter 1)
    default: 8150 => 8150 (adapter 1)
    default: 8151 => 8151 (adapter 1)
    default: 22 => 2222 (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
    default: Warning: Connection timeout. Retrying...
    default:
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default:
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if it's present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
GuestAdditions 4.3.30 running --- OK.
==> default: Checking for guest additions in VM...
==> default: Setting hostname...
==> default: Mounting shared folders...
    default: /vagrant => C:/Users/user01/Desktop/Vagrant_Files/razor_server
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` vagrant /vagrant
mount -t vboxsf -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant

The error output from the last command was:

mount: Protocol error
```

**Comment by predatorian3** [comment]
I just downgraded my version of VirtualBox to a known-working version that I use on my work laptop, version 4.3.20, and Vagrant still at 1.7.4. However, I'm still getting the same result. The only difference between the VM that I'm running on my work laptop versus the VM I'm trying to run here, is that my personal desktop is trying to use CentOS 6.7 as the guest, whereas my work laptop is using CentOS 6.6. 

Could that have something to do with it? When I downgraded VB versions, Vagrant recognized that, and installed the appropriate version of Guest Additions. However, I'm still getting the same protocol mismatch errors. Could this be something with CentOS 6.7?

**Comment by predatorian3** [comment]
I uninstalled Vagrant 1.7.4, Virtualbox 4.3.20, and downlaoded CentOS 6.6. I installed the Vagrant again, and the Virtualbox 4.3.30 again. Then I installed CentOS 6.6 in the VM. Then I installed all the development tools and guest additions and had to update the kernel, which updated me to 6.7 regardless. Then when I packaged it up, and tried vagrant up again, it actually worked. So, maybe it was the straight install to 6.7 instead of going from 6.6 to 6.7 was something different.
